common/core/overlaylayer.cpp

```diff
@@ -300,11 +307,22 @@ void OverlayLayer::InitializeState(HwcLayer* layer,
         display_frame_.left, display_frame_.right, display_frame_.top,
         display_frame_.bottom, left_source_constraint, left_constraint);
 
+    IMOSAICDISPLAYTRACE(
+        "Original surface_damage_ %d %d %d %d  left_source_constraint: %d "
+        "left_constraint: %d \n",
+        surface_damage_.left, surface_damage_.right, surface_damage_.top,
+        surface_damage_.bottom, left_source_constraint, left_constraint);
+
     display_frame_.bottom =
         std::min(max_height, static_cast<uint32_t>(display_frame_.bottom));
     display_frame_width_ = display_frame_.right - display_frame_.left;
     display_frame_height_ = display_frame_.bottom - display_frame_.top;
 
+    surface_damage_.left =
```

Comment:
Can you explain now surface_damage_ is in the scale of displayframe?
Why don't we need some scaling

Reply:
surface_damage_ and display_frame are in the same scale here. 
But the display_frame is re-calculated for the physical display in this function as 
display_frame_.left =
        (display_frame_.left - left_source_constraint) + left_constraint;
display_frame_.right =
        (display_frame_.right - left_source_constraint) + left_constraint;

so surface_damage_ need to be handled in the same way.

Reply:
@Shao-Feng 
OK.
I don't understand why this is a regression of your last partial update PR

Reply:
In last patch, we only transfer the surface_damage_ for partial refresh. before that patch, surface_damage will always be transferred, that is, below codes will be run on surface_damage even if the surface_damage_ is assigned with display_frame_. (In almost case, it is OK, because ox is 0. but in mosaic video case, it will make surface_damage be moved to right for half screen, and then be handled incorrectly.)
else if (transform_ == 0) {
      ox = display_frame_.left;
      oy = display_frame_.top;
      current_rendering_damage_.left = ox + translated_damage.left;
      current_rendering_damage_.top = oy + translated_damage.top;
      current_rendering_damage_.right = ox + translated_damage.right;
      current_rendering_damage_.bottom = oy + translated_damage.bottom;
    }
With previous patch, above code will only be invoked for partial refresh. the logical is correct. and the Rect of surface_damage_ and display_frame_ are correct too with the patch.